src/modules/powerrename/dll/dllmain.cpp

```diff
@@ -282,13 +283,15 @@ class PowerRenameModule : public PowertoyModuleIface
 
     void init_settings()
     {
-        m_enabled = CSettings::GetEnabled();
+        CSettingsInstance().LoadPowerRenameData();
+        m_enabled = CSettingsInstance().GetEnabled();

```

Comment:
This won't work, because you are caching the value, instead it needs to be read every time since PowerRename can enabled/disabled at any time in the settings.
Also, since it won't be acceptable to parse the whole json every time just for reading the `enable` state, until we find a better solution, we should keep using the registry for it.
ImageResizer has the same issue and we need to come up with the same solution for both.

Reply:
But `CSettings` object will exist no matter if `PowerRename` is disabled in the settings, and You can still read it from there. When we disable `PowerRename` we don't destroy if, but just put enabled flag and save settings. If we still need to be able to read this even when entire `PowerToys` applications is destroyed, than we need to change to registry. Otherway, this will work as long also `PowerToys` exist, no matter if `PowerRename` is enabled / disabled.

Reply:
The problem is that `CSettings` is caching the enabled value and if you update the settings in the UI the new value is not used by this code.
You have to not cache the enabled value and read it every time.

Reply:
We update json whenever we enable/disable from UI, also we update this "cached" value, so it's always up to date:
```
    void save_settings()
    {
        CSettingsInstance().SetEnabled(m_enabled);
        CSettingsInstance().SavePowerRenameData();
        Trace::EnablePowerRename(m_enabled);
    }
```

Reply:
It doesn't work because you don't read it again while the DLL is loaded in memory by Explorer. If you build the MSI, install it and test it, you'll see that it doesn't work.
Also, as I already said, since we should not parse the whole json every time we need to check the enabled value, we should not store it in json. ImageResizer has the same problem, we need to figure out a solution for both. In the meantime, please use the registry to persist the enabled state and make sure to read it every time it is used, as it was before this change.